src/live-reload.js

```diff
@@ -47,38 +32,78 @@ function reloadAllWindows() {
   return Promise.all(ret);
 }
 
-function enableLiveReloadNaive() {
-  let filesWeCareAbout = global.globalCompilerHost.listenToCompileEvents()
-    .filter(x => !FileChangedCache.isInNodeModules(x.filePath));
-
-  let weShouldReload = filesWeCareAbout
-    .mergeMap(x => watchPath(x.filePath).map(() => x))
-    .guaranteedThrottle(1*1000);
+function triggerHMRInRenderers() {
+  BrowserWindow.getAllWindows().forEach((window) => {
+    window.webContents.send('__electron-compile__HMR');
+  });
 
-  return weShouldReload
-    .switchMap(() => Observable.defer(() => Observable.fromPromise(reloadAllWindows()).timeout(5*1000).catch(() => Observable.empty())))
-    .subscribe(() => console.log("Reloaded all windows!"));
+  return Promise.resolve(true);
 }
 
-function triggerHMRInRenderers() {
+function triggerAssetReloadInRenderers(filePath) {
   BrowserWindow.getAllWindows().forEach((window) => {
-    window.webContents.send('__electron-compile__HMR');
+    window.webContents.send('__electron-compile__stylesheet_reload', filePath);
   });
 
   return Promise.resolve(true);
 }
 
-function enableReactHMR() {
-  global.__electron_compile_hmr_enabled__ = true;
+const defaultOptions = {
+  'strategy': {
+    'text/html': 'naive',
+    'text/tsx': 'react-hmr',
+    'text/jsx': 'react-hmr',
+    'application/javascript': 'react-hmr',
+    'text/stylus': 'hot-stylesheets',
+    'text/sass': 'hot-stylesheets',
+    'text/scss': 'hot-stylesheets',
+    'text/css' : 'hot-stylesheets'
+  }
+}
 
-  let filesWeCareAbout = global.globalCompilerHost.listenToCompileEvents()
-    .filter(x => !FileChangedCache.isInNodeModules(x.filePath));
+function setupWatchHMR(filePath) {
+  watchPath(filePath).subscribe(() => triggerHMRInRenderers())
+}
 
-  let weShouldReload = filesWeCareAbout
-    .mergeMap(x => watchPath(x.filePath).map(() => x))
-    .guaranteedThrottle(1*1000);
+function setWatchHotAssets(filePath) {
+  watchPath(filePath).subscribe(() => triggerAssetReloadInRenderers(filePath))
+}
 
-  return weShouldReload
-    .switchMap(() => Observable.defer(() => Observable.fromPromise(triggerHMRInRenderers()).catch(() => Observable.empty())))
-    .subscribe(() => console.log("HMR sent to all windows!"));
+function setupWatchNaive(filePath) {
+  watchPath(filePath).subscribe(() => reloadAllWindows())
 }
+
+export function enableLiveReload(options=defaultOptions) {
+  let { strategy } = options;
```

Comment:
Gotta handle the old syntax here (i.e. `{ strategy: 'naive' }`) or else it is a breaking change